app/main.py

```diff
@@ -1,2 +1,169 @@
+from typing import Union, Any, Iterable, Tuple
+
+
+# Linear Probing
 class Dictionary:
-    pass
+    def __init__(self,
+                 initial_capacity: int = 8,
+                 load_factor: float = 0.75) -> None:
+        # Cells will store (key, value) or None
+        self.table: list[Tuple[Any, Any]] = \
+            [None] * initial_capacity
+        self.size: int = 0
+        self.load_factor = load_factor
+        self.capacity = initial_capacity
+
+    def _hash(self, key: Any) -> int:
+        return hash(key) % self.capacity
+
+    def __setitem__(self, key: Any, value: Any) -> None:
+        index = self._hash(key)
+
+        # Linear probing: find an empty slot or replace the existing value
+        original_index = index
+        while self.table[index] is not None and self.table[index][0] != key:
+            index = (index + 1) % self.capacity
+            # We have returned to the starting point (table is full)
+            if index == original_index:
+                raise Exception("Dictionary is full")
+
+        # If the key already exists, update the value
+        if self.table[index] is not None:
+            self.table[index] = (key, value)
+        else:
+            self.table[index] = (key, value)
+            self.size += 1
+
+        # If the load factor exceeds the threshold, resize the table
+        if self.size / self.capacity > self.load_factor:
+            self._resize()
+
+    def __getitem__(self, key: Any) -> Any:
+        index = self._hash(key)
+        original_index = index
+
+        while self.table[index] is not None:
+            if self.table[index][0] == key:
+                return self.table[index][1]
+            index = (index + 1) % self.capacity
+            if index == original_index:
+                break
+
+        raise KeyError(f"Key '{key}' not found.")
+
+    def __len__(self) -> int:
+        return self.size
+
+    def __delitem__(self, key: Any) -> None:
+        index = self._hash(key)
+        original_index = index
+
+        while self.table[index] is not None:
+            if self.table[index][0] == key:
+                self.table[index] = None
+                self.size -= 1
+                # After removing an element,
+                # we need to rehash the remaining elements
+                self._rehash(index)
+                return
+            index = (index + 1) % self.capacity
+            if index == original_index:
+                break
+
+        raise KeyError(f"Key '{key}' not found.")
+
+    def __iter__(self) -> Iterable[Any]:
+        for item in self.table:
+            if item is not None:
+                yield item[0]
```

Comment:
The `__iter__` method currently yields keys instead of key-value pairs. If the intention is to iterate over keys, this is correct. However, if you want to iterate over key-value pairs, consider yielding `(key, value)` instead.